### PR TITLE
Data feed provider refactor

### PR DESCRIPTION
**Pull request by wladekb**
/cc @macbre

### CODE REVIEW

File: extensions/wikia/MyHome/SpecialWikiActivity.class.php [hunk 1]
```diff
@@ -117,6 +117,13 @@ function execute($par) {
 		global $wgHooks;
 		$wgHooks['PageHeaderIndexAfterExecute'][] = array($this, 'addNavigation');
 
+		if ($wgUser->isAnon()) {
+			$this->getOutput()->setSquidMaxage( 3600 ); // 1 hour
```
Comment (macbre) [inline extensions/wikia/MyHome/SpecialWikiActivity.class.php]: You can use `WikiaResponse::CACHE_STANDARD` instead.

**Comment by macbre** [comment]
:+1: for CDN caching for anons

File: extensions/wikia/MyHome/data/DataFeedProvider.php [hunk 2]
```diff
@@ -364,158 +386,192 @@ private function filterLog($res) {
...
-				}
+		if (!isset(self::$imageClassesCache[$imageName])) {
+			$imageClass = false;
+
+			$memcKey = wfMemcKey('DataFeedProvider','image_class', md5($imageName));
+
+			$imageLinkCount = $wgMemc->get($memcKey);
+			// Note that memcache returns null if record does not exists in cache
+			// versus 0 returned from database when image does not link to anything
+			if ($imageLinkCount === false) {
```
Comment (macbre) [inline extensions/wikia/MyHome/data/DataFeedProvider.php:548]: Let's store a number (`intval`) in memcache and check `is_numeric` here.

Comment (macbre) [inline extensions/wikia/MyHome/data/DataFeedProvider.php:548]: And is `$imageLinkCount` actually used below?

Comment (wladekb) [inline extensions/wikia/MyHome/data/DataFeedProvider.php:548]: Oops, I've seen this only after I clicked "merge"

**Pull request merged**